libs/vkd3d/state.c

```diff
@@ -593,58 +535,84 @@ static HRESULT d3d12_root_signature_init_root_descriptor_tables(struct d3d12_roo
             bool is_srv = range->RangeType == D3D12_DESCRIPTOR_RANGE_TYPE_SRV;
             bool is_uav = range->RangeType == D3D12_DESCRIPTOR_RANGE_TYPE_UAV;
 
-            vk_binding = d3d12_root_signature_assign_vk_bindings(root_signature,
-                    vkd3d_descriptor_type_from_d3d12_range_type(range->RangeType),
-                    range->RegisterSpace, range->BaseShaderRegister, range->NumDescriptors, false, is_uav, true,
-                    vkd3d_shader_visibility_from_d3d12(p->ShaderVisibility), context);
+            table->ranges[j].offset = range->OffsetInDescriptorsFromTableStart;
+            table->ranges[j].descriptor_count = range->NumDescriptors;
+            table->ranges[j].binding = context->vk_binding;
+            table->ranges[j].descriptor_magic = vkd3d_descriptor_magic_from_d3d12(range->RangeType);
+            table->ranges[j].base_register_idx = range->BaseShaderRegister;
+            table->ranges[j].register_space = range->RegisterSpace;
+
+            if (range->OffsetInDescriptorsFromTableStart != D3D12_DESCRIPTOR_RANGE_OFFSET_APPEND)
+                range_descriptor_offset = range->OffsetInDescriptorsFromTableStart;
+
+            binding.type = vkd3d_descriptor_type_from_d3d12_range_type(range->RangeType);
+            binding.register_space = range->RegisterSpace;
+            binding.register_index = range->BaseShaderRegister;
+            binding.register_count = range->NumDescriptors;
+            binding.descriptor_table = table->table_index;
+            binding.descriptor_offset = range_descriptor_offset;
+            binding.shader_visibility = vkd3d_shader_visibility_from_d3d12(p->ShaderVisibility);
+            binding.binding.set = context->vk_set;
+            binding.binding.binding = context->vk_binding;
 
             /* Unroll descriptor range. */
             for (k = 0; k < range->NumDescriptors; ++k)
             {
-                uint32_t vk_current_binding = vk_binding + k;
-
+                VkDescriptorSetLayoutBinding vk_binding;
```

Comment:
Why unroll when we have descriptor_count now propagated to shader compiler?

Reply:
We do not support indexing on packed descriptors in the shader compiler.

Reply:
Ok, so all SRVs currently enter this path. I assume the idea is that further commits will only enter this loop for UAV/CBV?

Reply:
Yes, bindless doesn't use this code path.